src/background-plugin.ts

```diff
@@ -192,16 +182,45 @@ class BackgroundPlugin {
     return data;
   }
 
+  async getExports(name?: string): Promise<WebAssembly.ModuleExportDescriptor[]> {
+    return await this.#invoke('getExports', name ?? '0');
+  }
+
+  async getImports(name?: string): Promise<WebAssembly.ModuleImportDescriptor[]> {
+    return await this.#invoke('getImports', name ?? '0');
+  }
+
+  async getInstance(): Promise<WebAssembly.Instance> {
+    throw new Error('todo');
+  }
+
+  async close(): Promise<void> {
+    if (this.worker) {
+      this.worker.terminate();
+      this.worker = null as any;
+    }
+  }
+
   // guest -> host invoke()
   async #handleInvoke(ev: any) {
     const writer = new RingBufferWriter(this.sharedData);
     const namespace = this.opts.functions[ev.namespace];
     const func = (namespace ?? {})[ev.func];
+    // XXX(chrisdickinson): this is cürsëd code. Add a setTimeout because some platforms
+    // don't spin their event loops if the only pending item is a Promise generated by Atomics.waitAsync.
+    //
+    // - https://github.com/nodejs/node/pull/44409
+    // - https://github.com/denoland/deno/issues/14786
+    const timer = setInterval(() => {}, 0);
```

Comment:
This code moved – the timer used to be created per `AtomicsWaitAsync`, now we create it once at the start of the invocation and run it until we're done with the invocation.